libsolidity/analysis/DeclarationContainer.cpp

```diff
@@ -115,3 +116,23 @@ std::vector<Declaration const*> DeclarationContainer::resolveName(ASTString cons
 		return m_enclosingContainer->resolveName(_name, true);
 	return vector<Declaration const*>({});
 }
+
+vector<ASTString> DeclarationContainer::similarNames(ASTString const& _name) const
+{
+	vector<ASTString> similar;
+
+	for (auto const& declaration: m_declarations)
+	{
+		string const& declarationName = declaration.first;
+		if (stringWithinDistance(_name, declarationName, MAXIMUM_DISTANCE))
+			similar.push_back(declarationName);
+	}
+
+	if (m_enclosingContainer)
+	{
+		vector<ASTString> enclosingSimilar = m_enclosingContainer->similarNames(_name);
```

Comment:
We have an operator overload that allows `similar += m_enclosingContainer->similarNames(_name);`